### PR TITLE
Improve UI video teachers dashboard

### PR DESCRIPTION
**Pull request by quitterie-lcs**
# Purpose

Feedbacks from the learning analytics workshop concerned the user experience that needs to be improved in the teachers dashboard

# Proposal

- [x] Fix following variable values: Statements interval = 1d / View count threshold = 30s / Event group interval = 1
- [x] Add the title, the number of students and the start and end dates of the course in the course overview

### CODE REVIEW

**Comment by quitterie-lcs** [comment]
@jmaupetit @SergioSim Maybe the three constants variables can be just regraded as project constants and not variables anymore? WDYT?

**Comment by SergioSim** [comment]
> 

Hello @quitterie-lcs ) If we limit the user choice to a single value, I think it would be better to have them as constants)

File: src/dashboards/teachers/course.jsonnet [hunk 1]
```diff
@@ -15,40 +17,110 @@ dashboard.new(
...
-.addTemplate(teachers_common.templates.view_count_threshold)
-.addTemplate(teachers_common.templates.statements_interval)
+.addTemplate(teachers_common.templates.title)
+.addTemplate(teachers_common.templates.start_date)
+.addTemplate(teachers_common.templates.end_date)
+.addPanel(
+  text.new(
+    title='Course',
+    content=|||
+      # Welcome to ${TITLE} course!
```
Comment (jmaupetit) [inline src/dashboards/teachers/course.jsonnet:32]:
```suggestion
      # ${TITLE}
```

File: src/dashboards/teachers/course.jsonnet [hunk 2]
```diff
@@ -15,40 +17,110 @@ dashboard.new(
...
+        },
+      ],
+      timeField='@timestamp'
+    )
+  ),
+  gridPos={ x: 0, y: 9, w: 4, h: 6 }
+)
+.addPanel(
+  statPanel.new(
+    title='Learners',
```
Comment (jmaupetit) [inline src/dashboards/teachers/course.jsonnet:69]:
```suggestion
    title='Enrollments',
```

File: fixtures/mysql/edxapp.sql [hunk 3]
```diff
@@ -13,9 +16,23 @@ CREATE TABLE IF NOT EXISTS auth_user (
...
-INSERT INTO student_courseaccessrole (id, `user_id`, `course_id`, role) VALUES (1, 1, 'course-v1:FUN-MOOC+00001+session01', 'staff');
-INSERT INTO student_courseaccessrole (id, `user_id`, `course_id`, role) VALUES (2, 1, 'course-v1:FUN-MOOC+00002+session01', 'instructor');
+CREATE TABLE IF NOT EXISTS student_courseenrollment (
+    id INT PRIMARY KEY NOT NULL,
+    `user_id` INT,
+    `course_id` VARCHAR(255),
+    `is_active` INT
+);
+INSERT INTO courses_course (id, `key`, `title`, `start_date`, `end_date`) VALUES (1, 'course-v1:FUN-MOOC+00001+session01', 'FUN Mooc Course 1 Session 1', DATE('2000-01-01'), DATE('2030-01-01'));
+INSERT INTO courses_course (id, `key`, `title`, `start_date`, `end_date`) VALUES (2, 'course-v1:FUN-MOOC+00002+session01', 'FUN Mooc Course 2 Session 2', DATE('2000-01-01'), DATE('2030-01-01'));
```
Comment (jmaupetit) [inline fixtures/mysql/edxapp.sql:26]: Ideally, course dates should cover the same period as ES data, _i.e._ now - 4 weeks?

File: src/dashboards/teachers/course.jsonnet [hunk 4]
```diff
@@ -15,40 +17,110 @@ dashboard.new(
...
+    )
+  ),
+  gridPos={ x: 4, y: 9, w: 4, h: 6 }
+)
+.addPanel(
+  text.new(
+    title='Dates',
+    content=|||
+      ## From ${START_DATE} 
+      ## To ${END_DATE}
```
Comment (jmaupetit) [inline src/dashboards/teachers/course.jsonnet:90]:
```suggestion
      ## Started: ${START_DATE} 
      ## Ended: ${END_DATE}
```

File: src/dashboards/teachers/common.libsonnet [hunk 5]
```diff
@@ -70,18 +90,40 @@ local common = import '../common.libsonnet';
...
+    ),
+    end_date: template.new(
+      name='END_DATE',
+      current='all',
+      label='End Date',
+      datasource=common.datasources.edx_app,
+      query=$.queries.course_end_date,
+      hide='variable',
+      refresh='time'
+    ),
```
Comment (jmaupetit) [inline src/dashboards/teachers/common.libsonnet:119]: Maybe we can set the dashboard default analysis period to those values (start/end dates)?

**Review by jmaupetit — COMMENTED** [review]
Don't forget the CHANGELOG! :wink:

**Comment by jmaupetit** [comment]
I've added a commit with minor fix and improvements to the fixtures script. Feel free to do whatever you want with it!

File: scripts/users-permissions.sh [hunk 6]
```diff
@@ -13,9 +13,8 @@ TEACHER_TEAM_ID=$(${curl} -d "name=Teacher" grafana:3000/api/teams | jq '.teamId
 STUDENT_USER_ID=$(${curl} -d "email=student@example.org&login=student&password=funfunfun" grafana:3000/api/admin/users | jq '.id')
 TEACHER_USER_ID=$(${curl} -d "email=teacher@example.org&login=teacher&password=funfunfun" grafana:3000/api/admin/users | jq '.id')
 
-# Return id of the "videos" folder 
-# FIXME: to be updated to "teachers" folder
-VIDEO_FOLDER_UID=$(${curl} "http://grafana:3000/api/folders" | jq -r '.[] | select(.title=="videos") | .uid')
+# Return id of the "videos" folder
```
Comment (jmaupetit) [inline scripts/users-permissions.sh:16]:
```suggestion
# Return id of the "teachers" folder
```

**Review by jmaupetit — APPROVED** [review]
A tiny fix and we are done! :muscle:

**Pull request merged**